scripts/automated_ingestion/eessi_task.py

```diff
@@ -0,0 +1,1201 @@
+        repo_name = self.description.get_repo_name()
+        pr_number = self.description.get_pr_number()
+        pr_url = f"https://github.com/{repo_name}/pull/{pr_number}"
+        seq_num = self._determine_sequence_number_from_pull_request_directory()
+
+        self._create_task_summary()
+        contents_overview = self._create_pr_contents_overview()
+        pr_body_format = self.config["github"]["grouped_pr_body"]
+        pr_body = pr_body_format.format(
+            cvmfs_repo=self.cvmfs_repo,
+            pr=pr_number,
+            pr_url=pr_url,
+            repo=repo_name,
+            seq_num=seq_num,
+            contents=contents_overview,
+            analysis="<dd>TO BE DONE</dd>",
+            action="<dd>TO BE DONE</dd>",
+        )
+        pull_request.edit(body=pr_body)
+
+        log_message(LoggingScope.TASK_OPS, "INFO", "PR updated: '%s'", pull_request)
+
+    @log_function_entry_exit()
+    def _handle_add_payload_staged(self):
+        """Handler for ADD action in PAYLOAD_STAGED state"""
+        log_message(LoggingScope.TASK_OPS, "INFO", "Handling ADD action in PAYLOAD_STAGED state: '%s'",
+                    self.description.get_task_file_name())
+        next_state = self._next_state(EESSITaskState.PAYLOAD_STAGED)
+        approved_state = EESSITaskState.APPROVED
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                    "next_state: '%s', approved_state: '%s'", next_state, approved_state)
+
+        default_branch_name = self.git_repo.default_branch
+        default_branch = self._get_branch_from_name(default_branch_name)
+        default_sha = default_branch.commit.sha
+        feature_branch_name = self._determine_feature_branch_name()
+        feature_branch = self._get_branch_from_name(feature_branch_name)
+        if not feature_branch:
+            # feature branch does not exist
+            # TODO: could have been merged already --> check if PR corresponding to the feature branch exists
+            # ASSUME: it has not existed before --> create it
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "branch '%s' does not exist, creating it", feature_branch_name)
+
+            feature_branch = self.git_repo.create_git_ref(f"refs/heads/{feature_branch_name}", default_sha)
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "branch '%s' created: '%s'", feature_branch_name, feature_branch)
+        else:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "found existing branch for '%s': '%s'", feature_branch_name, feature_branch)
+
+        pull_request = self._find_pr_for_branch(feature_branch_name)
+        if not pull_request:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "no PR found for branch '%s'", feature_branch_name)
+
+            # TODO: add failure handling (capture result and act on it)
```

Comment:
```suggestion
            # The default branch should reflect the 'current' state. I.e. there, we write next_state into the state file
            # In the feature branch, we write approved_state into the state file so that if-and-when the PR is merged
            # into the default branch, the default branch reflects the approval of the PR.
            # TODO: add failure handling (capture result and act on it)
```